utils/redisCache.ts

```diff
@@ -0,0 +1,60 @@
+import { connect, Redis } from "https://deno.land/x/redis@v0.29.0/mod.ts";
+import { WeakLRUCache } from "https://deno.land/x/weakcache@v1.1.4/index.js";
+
+const fallbackCache = new WeakLRUCache();
+
+class RedisCache {
+  redisPromise: Promise<Redis | void>;
+  redis: Redis;
+  isResolved = true;
+  constructor() {
+    this.redisPromise = Promise.resolve();
+    this.redis = null as unknown as Redis;
+    this.#connect();
+  }
+
+  #connect = () => {
+    if (this.isResolved && (this.redis === null || this.redis.isClosed)) {
```

Comment:
you might want a single flight here as you cannot guarantee a condition-free on connect